akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2Demux.scala

```diff
@@ -357,21 +361,40 @@ private[http2] abstract class Http2Demux(http2Settings: Http2CommonSettings, ini
         }
       })
 
+      // -----------------------------------------------------------------
       // FIXME: What if user handler doesn't pull in new substreams? Should we reject them
-      //        after a while or buffer only a limited amount? We should also be able to
-      //        keep the buffer limited to the number of concurrent streams as negotiated
-      //        with the other side.
-      val bufferedSubStreamOutput = new BufferedOutlet[Http2SubStream](substreamOut)
+      //        after a while or buffer only a limited amount?
+      substreamOut.setHandler(new OutHandler() {
+        // This is a noop handler because we're never pushing on this outlet and,
+        // instead, we always emit (which teporarily swaps this handler with an
+        // internal handler that implements the emit.
+        override def onPull(): Unit = ()
+      })
       override def dispatchSubstream(initialHeaders: ParsedHeadersFrame, data: Source[Any, Any], correlationAttributes: Map[AttributeKey[_], _]): Unit =
-        bufferedSubStreamOutput.push(Http2SubStream(initialHeaders, data, correlationAttributes))
+        emit(substreamOut, Http2SubStream(initialHeaders, data, correlationAttributes))
 
```

Comment:
This is the key change: we're dropping `BufferedOutlet` in favor of using the internal buffer `emit` provides. 

In order to use `emit` we're setting a noop handler (since `emit` fails to operate on outlets without handler).